openmc/mgxs_library.py

```diff
@@ -970,109 +1103,189 @@ def set_multiplicity_mgxs(self, nuscatter, scatter=None, nuclide='total',
+            xsgrp = grp.create_group(str(int(np.round(temperature))) + "K")
+            if self._total[i] is None:
+                raise ValueError('total data must be provided when writing '
+                                 'the HDF5 library')
+            xsgrp.create_dataset("total", data=self._total[i],
+                                 compression=compression)
+            if self._absorption[i] is None:
+                raise ValueError('absorption data must be provided when '
+                                 'writing the HDF5 library')
+            xsgrp.create_dataset("absorption", data=self._absorption[i],
+                                 compression=compression)
+            if self.fissionable:
+                if self._fission[i] is not None:
+                    xsgrp.create_dataset("fission", data=self._fission[i],
+                                         compression=compression)
+                if self._kappa_fission[i] is not None:
+                    xsgrp.create_dataset("kappa-fission",
+                                         data=self._kappa_fission[i],
+                                         compression=compression)
+                if self._chi[i] is not None:
+                    xsgrp.create_dataset("chi", data=self._chi[i],
+                                         compression=compression)
+                if self._nu_fission[i] is None:
+                    raise ValueError('nu-fission data must be provided when '
+                                     'writing the HDF5 library')
+                xsgrp.create_dataset("nu-fission",
+                                     data=self._nu_fission[i],
+                                     compression=compression)
+
+            if self._scatter_matrix[i] is None:
+                raise ValueError('Scatter matrix must be provided when '
+                                 'writing the HDF5 library')
+            # Get the sparse scattering data to print to the library
+            G = self.energy_groups.num_groups
+            if self.representation == 'isotropic':
+                g_out_bounds = np.zeros((G, 2), dtype=np.int)
+                for g_in in range(G):
+                    nz = np.nonzero(self._scatter_matrix[i][0, g_in, :])
+                    g_out_bounds[g_in, 0] = nz[0][0]
+                    g_out_bounds[g_in, 1] = nz[0][-1]
+                # Now create the flattened scatter matrix array
+                matrix = self._scatter_matrix[i]
+                flat_scatt = []
+                for g_in in range(G):
+                    for g_out in range(g_out_bounds[g_in, 0],
+                                       g_out_bounds[g_in, 1] + 1):
+                        for l in range(len(matrix[:, g_in, g_out])):
+                            flat_scatt.append(matrix[l, g_in, g_out])
+                # And write it.
+                scatt_grp = xsgrp.create_group('scatter data')
+                scatt_grp.create_dataset("scatter matrix",
```

Comment:
I like the readability a bit more here with the underscore before `grp`. Could you add an underscore in `xsgrp` above?